File: Java-BackEnd/JansiSnack_BackEnd/Order/src/main/java/tech/wetech/order/producer/model/OrderForCheck.java
package tech.wetech.order.producer.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

/**
 * @author Jansonci
 */
@Data
public class OrderForCheck {
  Long orderId;

 private List<OrderItemEntry> orderItems;

  @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
  OffsetDateTime payTime;

  public OrderForCheck() {
  }

  public OrderForCheck(Long orderId, List<OrderItemEntry> orderItem, OffsetDateTime payTime, Long stillNeeds, BigDecimal amount, String address) {
    this.orderId = orderId;
    this.orderItems = orderItem;
    this.payTime = payTime;
    this.stillNeeds = stillNeeds;
    this.amount = amount;
    this.address = address;
  }

  Long stillNeeds;

  BigDecimal amount;

  String address;
}

File: Kotlin-Android/JansiSnack_Front/app/src/main/java/com/example/jetsnack/model/OrderForCheck.kt
package com.example.jetsnack.model

import android.os.Build
import androidx.annotation.RequiresApi
import java.math.BigDecimal
import java.time.OffsetDateTime

data class OrderForCheck(
    val orderId: Long,
    val orderItems: List<OrderItemEntry>,  // 修改此处为列表
    val payTime: OffsetDateTime,           // 注意这里使用OffsetDateTime而不是OffsetTime
    val stillNeeds: Long,
    val amount: BigDecimal,
    val address: String
)

data class OrderItemEntry(
    val dessert: Dessert,  // 假设Dessert是一个已定义的数据类
    val quantity: Int
)

@RequiresApi(Build.VERSION_CODES.O)
val orderForCheck: OrderForCheck = OrderForCheck(1, listOf(
    OrderItemEntry(
        Dessert(1,"Cupcake","","","",1.00,"",
    1,1,"","","",1,1.00,"https://source.unsplash.com/3U2V5WqK1PQ","", listOf()), 3)
), OffsetDateTime.parse("2024-05-11T16:03:37+08:00"), 2, 20.00.toBigDecimal(),"")